src/main/java/io/zksync/signer/EthSigner.java

```diff
@@ -1,120 +1,102 @@
-            Sign.signedPrefixedMessageToKey(message.getBytes(), signatureData) :
-            Sign.signedMessageToKey(message.getBytes(), signatureData);
-        return credentials.getEcKeyPair().getPublicKey().equals(publicKey);
-    }
-
-    private static Credentials generateCredentialsFromMnemonic(String mnemonic, int accountIndex) {
-        //m/44'/60'/0'/0 derivation path
-        int[] derivationPath = {44 | Bip32ECKeyPair.HARDENED_BIT, 60 | Bip32ECKeyPair.HARDENED_BIT, 0 | Bip32ECKeyPair.HARDENED_BIT, 0,accountIndex};
+import org.web3j.crypto.Credentials;
 
-        // Generate a BIP32 master keypair from the mnemonic phrase
-        Bip32ECKeyPair masterKeypair = Bip32ECKeyPair.generateKeyPair(
-                MnemonicUtils.generateSeed(mnemonic, ""));
+import io.zksync.domain.token.Token;
 
-        // Derive the keypair using the derivation path
-        Bip32ECKeyPair  derivedKeyPair = Bip32ECKeyPair.deriveKeyPair(masterKeypair, derivationPath);
+public interface EthSigner {
+    
+    /**
+     * Get wallet address
+     * 
+     * @return Address in hex string
+     */
+    String getAddress();
+
+    /**
+     * Get internal credentials. @see org.web3j.crypto.Credentials
+     * 
+     * @return Credentials object that contains private and public key pair
+     */
+    Credentials getCredentials();
+
+    /**
+     * Sign ZkSync `ChangePubKey` operation message
+     * 
+     * @param pubKeyHash - ZkSync public key hash
+     * @param nonce - Nonce of the account
+     * @param accountId - Id of the account in ZkSync
+     * @return Signature object
+     */
+    EthSignature signChangePubKey(String pubKeyHash, Integer nonce, Integer accountId);
+
+    /**
+     * Sign ZkSync `Transfer` operation message
+     * 
+     * @param to - Address of receiver
+     * @param accountId - Id of the sender account in ZkSync
+     * @param nonce - Nonce of the sender account
+     * @param amount - Amount tokens to transfer
+     * @param token - Token object supported by ZkSync
+     * @param fee - Cost of transaction in ZkSync network
+     * @return Signature object
+     */
+    EthSignature signTransfer(String to, Integer accountId, Integer nonce, BigInteger amount, Token token, BigInteger fee);
+
+    /**
+     * Sign ZkSync `Withdraw` operation message
+     * 
+     * @param to - Address of receiver
+     * @param accountId - Id of the account in ZkSync
+     * @param nonce - Nonce of the account
+     * @param amount - Amount tokens to withdraw
+     * @param token - Token object supported by ZkSync
+     * @param fee - Cost of transaction in ZkSync network
+     * @return Signature object
+     */
+    EthSignature signWithdraw(String to, Integer accountId, Integer nonce, BigInteger amount, Token token, BigInteger fee);
+
+    /**
+     * Sign raw message
+     * 
+     * @param message - Message to sign
+     * @return Signature object
+     */
+    EthSignature signMessage(String message);
```

Comment:
All the signing functions should be async, since the JSON-RPC signers, which will be implemented in the future are remote.